### ISSUE TITLE
Need to close window every time to proceed ahead with the epoch.

### ISSUE THREAD

**Issue by ravishk1** [stared/livelossplot#76]
Training stops after every epoch completion, need to manually close the window and then only it proceeds with next epoch. Can't it simultaneously plot and proceed to the next epoch?

Thanks for this library btw. :)

**Comment by stared** [comment]
@ravishk1 Do you run it within Jupyter Notebook? 

(As of now, it is not suited for running it from other environments.)

**Comment by PhilMacKay** [comment]
Hello! If we modify the file `livelossplot/core.py` on line 30 to specify a figure name (for example, it could be "livelossplot"!), add a `plt.clf()` right after line 30, and add the keyword `block=False` in the call to `plt.show()` on line 71, it would be possible to use Livelossplot from IPython or from a Python script!

I have been using these modifications locally with the Qt5Agg backend for a while and it works well. The call to `plt.clf()` is somewhat slow, but if an epoch lasts more than a full second, it's not noticeable.

I would gladly make a pull-request if you're interested, I think these small modifications would solve a large part of non-Notebook support!

**Comment by stared** [comment]
Thanks for your comment, @PhilMacKay. I had some time off with livelossplot. But in the last two weeks, we rewrote the library (with @Bartolo1024) and released as 0.5.0. Now it is more modular and it will be easier to make such changes.

As you see, we started targetting non-Jupyter environments, see  https://github.com/stared/livelossplot/blob/master/livelossplot/outputs/. We will try to include it in 0.5.1 release, most likely using (for livelossplot) parameters to specify if it is `'jupyter'`, `'script'` or `'server'` (only saving plots as files).  PRs are welcome! :)
